types/array_basic.go

```diff
@@ -15,13 +14,18 @@ const BasicArraySizeCache = 100000
 var fastSumHashKey = toBytes32([]byte("hash_fast_sum64_key"))
 
 type basicArraySSZ struct {
-	hashCache *ccache.Cache
+	hashCache *ristretto.Cache
 	lock      sync.Mutex
 }
 
 func newBasicArraySSZ() *basicArraySSZ {
+	cache, _ := ristretto.NewCache(&ristretto.Config{
+		NumCounters: BasicArraySizeCache, // number of keys to track frequency of (1M).
+		MaxCost:     1 << 23,             // maximum cost of cache (10MB).
```

Comment:
Aren't state objects 800M? Why is this so low?

Reply:
That would only be for mainnet, currently it would be 60KB. And this only caches basic arrays.

Reply:
The problem here is that we aren't costing the insertions. You've set cost = 0 for every set. Will we ever hit MaxCost?